error/src/util.rs

```diff
@@ -35,7 +56,26 @@ macro_rules! impl_error_conversion_with_kind {
     };
 }
 
-/// TODO(doc): @keroro520
+/// A macro to implement conversion from source type to target type based on an implicit middle adaptor.
+///
+/// ## Examples
+///
+/// ```rust
+/// impl_error_conversion_with_adaptor!(SourceType, AdaptorType, TargetType)
```

Comment:
```suggestion
/// ```ignore
/// impl_error_conversion_with_adaptor!(SourceType, AdaptorType, TargetType);
```